client/src/components/overview/Cart.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect } from 'react';
+
+const Cart = ( props ) => {
+  // const [cartItems, serCurrentCart] = await useState({}) need to pull a user's cart
+  const [isLoading, setLoading] = useState(false)
+  const [styleId, setStyleId] = useState('');
+  const [styleList, setStyleList] = useState([])
+  const [selectedStyle, setStyle] = useState({});
+  const [skus, setSKUs] = useState([]);
+  const [currentSKU, setSKU] = useState({});
+  const [selectedSize, setSize] = useState('');
+  const [quantities, setQuantities] = useState([]);
+  const [selectedQuantity, setQuantity] = useState('');
+
+  useEffect( () => {
+    setLoading(true);
+    setStyleList(props.styleList);
+    setStyle(props.style);
+    setStyleId(selectedStyle.style_id);
+    setLoading(false);
+  }, [])
+
+  useEffect( () => {
+    setSKUs(parseSKUs(selectedStyle));
+  }, [selectedStyle])
+
+  useEffect( () => {
+    for (var i = 0; i < skus.length; i++) {
+      if (Object.values(skus[i]).includes(selectedSize)) {
+        setSKU(skus[i]);
+        var arrayOfQuantities = []
+        for (var j = 0; j < skus[i].quantity; j++) {
+          arrayOfQuantities.push(j + 1)
+        }
+      }
+    }
+    setQuantities(arrayOfQuantities);
+
+  }, [selectedSize])
+
+  // useEffect( () => {
+
+  // }, [currentSKU])
+
+
+
+  function handleStyleClick(value) {
+    let list = styleList;
+    for (var i = 0; i < list.length; i++) {
+      if (list[i].style_id === Number(value)){
+        setStyle(list[i]);
+        setSKUs(parseSKUs(list[i]));
```

Comment:
Isn't this redundant with the effect on line 23? setSKUs is going to run after we set the style.

Reply:
Yup you're right. Good catch. Will edit it out. Missed it when i was refactoring the onEffect to load one after another.